test/Microsoft.AspNetCore.Razor.Evolution.Test/IntegrationTests/RazorIRNodeWriter.cs

```diff
@@ -155,9 +156,10 @@ protected void WriteContent(string content)
                 return;
             }
 
-            // We explicitly escape newlines in node content so that the IR can be compared line-by-line.
+            // We explicitly escape newlines in node content so that the IR can be compared line-by-line. The escaped
+            // newline cannot be platform specific so we need to replace it with a custom identifier.
             // Also, escape our separator so we can search for ` - `to find delimiters.
-            _writer.Write(content.Replace("\r", "\\r").Replace("\n", "\\n").Replace("-", "\\-"));
+            _writer.Write(content.Replace(Environment.NewLine, "[NEWLINE]").Replace(" - ", "\\-"));
```

Comment:
Why not just normalize it to `\\n` - We have code elsewhere that does this already, it replaces `\r` with `""` (empty)